.github/workflows/cherry-pick.yml

```diff
@@ -0,0 +1,106 @@
+    runs-on: ubuntu-latest
+    name: Collect target branches from Pull Request
+    outputs:
+      cherry_pick_target_branches: ${{ steps.target_branches.outputs.cherry_pick_target_branches }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      # Collect target branches in a list. The list includes:
+      # - default target branch
+      # - any branch specified in PR comments with the syntax
+      #     /cherry-pick <branch-1> <branch-2>
+      # The final list will only contain unique branches
+      - name: Create list of target branches
+        id: target_branches
+        run: |
+          cherry_pick_branches_json=$(gh pr view ${{ github.event.number }} \
+                                       --repo ${{ github.repository }} \
+                                       --json comments \
+                                       --jq '.comments
+                                              | map(select(.authorAssociation == "MEMBER" or .authorAssociation == "OWNER")
+                                                    | .body
+                                                    | select(startswith("/cherry-pick "))
+                                                    | ltrimstr("/cherry-pick ")
+                                                    | split(" "))
+                                              | flatten
+                                              | . += ["${{ fromJSON(env.CHERRY_PICK_BRANCH_MAPPING)[github.base_ref] }}"]
+                                              | unique')
+          branches_list=$(echo $cherry_pick_branches_json | jq -r @sh | tr -d "'")
+          IFS=" " read -ra target_branches <<< "$branches_list"
+          echo "cherry_pick_target_branches=${target_branches[*]}" >> $GITHUB_OUTPUT
+
+  # Actual cherry-pick work for each target branch
+  cherry_pick_commit:
+    needs: collect_target_branches
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    name: Cherry pick
+    steps:
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+          fetch-depth: 0
+
+      - name: Configure local git repo
+        run: |
+          git config --local user.email "cblmargh@microsoft.com"
+          git config --local user.name "CBL-Mariner Servicing Account"
+
+      # Run cherry-pick for each target branch and create new PR for it. If cherry-pick fails,
+      # output conflicts to the action logs and notify in the original PR.
+      - name: Run cherry-pick action
+        run: |
+          for target_branch in ${{ needs.collect_target_branches.outputs.cherry_pick_target_branches }}; do
+          toolkit/scripts/cherry_pick.sh \
+              -r "${{ github.repository }}" \
+              -p "${{ github.event.pull_request.number }}" \
+              -t "$target_branch" || true
```

Comment:
Why the `|| true`? Shouldn't we fail in case of an error?

Reply:
The error handling is done inside the script. I want to continue the cherry-pick work for other branches even if one fails, regardless of the reason. Probably need to check for the exit code to fail the workflow run later

Reply:
And if we want to continue, please add a comment explaining why this is the preference. This may help a future developer looking at this piece of code.

Reply:
Your approach makes sense. And I agree, that we should save the failure information to fail the whole action in the end.